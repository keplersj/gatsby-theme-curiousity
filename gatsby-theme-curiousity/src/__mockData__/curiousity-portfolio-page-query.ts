import { PortfolioPageQuery } from "../templates/portfolio-query";

export const CuriousityPortfolioPageQuery: PortfolioPageQuery = {
  allPortfolioItem: {
    edges: [{ node: { id: "", excerpt: "", slug: "", title: "" } }],
  },
};
