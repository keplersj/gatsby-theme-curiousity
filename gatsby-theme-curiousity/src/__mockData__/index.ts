export * from "./CuriosityPortfolioPieceItemQuery";
export * from "./CuriosityPostQuery";
export * from "./CuriousityBaseLayoutData";
export * from "./CuriousityPiecePage";
export * from "./CuriousityPiecePageQuery";
export * from "./CuriousityPortfolioPageQuery";
