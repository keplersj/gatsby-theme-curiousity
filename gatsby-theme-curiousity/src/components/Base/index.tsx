import * as React from "react";
import { Helmet } from "react-helmet";
import { useStaticQuery, graphql } from "gatsby";

import "modern-normalize";
import "starstuff-style";

export interface BaseLayoutData {
  site: {
    siteMetadata: {
      title: string;
      siteUrl: string;
    };
  };
}

interface Props {
  title?: string;
  description?: string;
}

const BaseLayout = (
  properties: React.PropsWithChildren<Props>
): React.ReactElement<React.PropsWithChildren<Props>> => {
  const data = useStaticQuery<BaseLayoutData>(graphql`
    query CuriousityBaseLayoutData {
      site {
        siteMetadata {
          title
          siteUrl
        }
      }
    }
  `);

  return (
    <main>
      <Helmet
        titleTemplate={`%s | ${data.site.siteMetadata.title}`}
        defaultTitle={data.site.siteMetadata.title}
      >
        <html lang="en" />
        {properties.title && <title>{properties.title}</title>}
        {properties.description && (
          <meta name="description" content={properties.description} />
        )}
        <meta
          property="og:title"
          content={
            (properties.title &&
              `${properties.title} | ${data.site.siteMetadata.title}`) ||
            data.site.siteMetadata.title
          }
        />
        <meta property="og:url" content={data.site.siteMetadata.siteUrl} />
        {properties.description && (
          <meta property="og:description" content={properties.description} />
        )}
      </Helmet>

      {properties.children}
    </main>
  );
};

export default BaseLayout;
