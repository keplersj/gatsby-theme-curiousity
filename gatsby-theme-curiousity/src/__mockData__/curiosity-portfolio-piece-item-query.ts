import { PortfolioPieceItemQuery } from "../components/PortfolioListItem";

export const CuriosityPortfolioPieceItemQuery: PortfolioPieceItemQuery = {
  site: {
    siteMetadata: {
      siteUrl: "https://example.dev/"
    }
  }
};
