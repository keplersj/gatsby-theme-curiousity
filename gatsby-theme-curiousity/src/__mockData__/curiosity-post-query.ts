import { PostQuery } from "../components/Piece";

export const CuriosityPostQuery: PostQuery = {
  site: {
    siteMetadata: {
      siteUrl: "https://example.dev/",
    },
  },
};
