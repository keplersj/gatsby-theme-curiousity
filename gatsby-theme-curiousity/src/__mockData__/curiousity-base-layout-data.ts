import { BaseLayoutData } from "../components/Base";

export const CuriousityBaseLayoutData: BaseLayoutData = {
  site: {
    siteMetadata: {
      title: "Test Site",
      siteUrl: "https://example.dev/"
    }
  }
};
