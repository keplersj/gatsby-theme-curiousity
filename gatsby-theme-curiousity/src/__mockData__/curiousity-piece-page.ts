import { PiecePage } from "../components/Piece";

export const CuriousityPiecePage: PiecePage = {
  id: "42",
  excerpt: "test excerpt",
  html: "<div><span>Test Piece</span></div>",
  slug: "/test-piece",
  title: "Test Piece",
  tags: [],
  keywords: [],
  metadata: {
    type: ["Test"],
    status: ["Test"],
    role: ["Test"],
    homepage: "https://example.dev/",
    rubygemsGemName: "test",
    npmPackageName: "test",
    githubRepo: "test/test",
  },
  fileRelativePath: "",
  rawFrontmatter: "",
  rawMarkdownBody: "",
  frontmatter: {},
};
