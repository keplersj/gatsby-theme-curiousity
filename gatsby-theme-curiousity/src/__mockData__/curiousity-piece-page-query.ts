import { CuriousityPiecePage } from "./curiousity-piece-page";
import { PiecePageQuery } from "../templates/piece-query";

export const CuriousityPiecePageQuery: PiecePageQuery = {
  portfolioItem: CuriousityPiecePage,
};
