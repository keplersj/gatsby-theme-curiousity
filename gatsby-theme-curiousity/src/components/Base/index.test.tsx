import * as React from "react";
import renderer from "react-test-renderer";
import { useStaticQuery } from "gatsby";
import Base from "./";
import { CuriousityBaseLayoutData } from "../../__mockData__";

describe("Base Layout", () => {
  beforeEach(() => {
    (useStaticQuery as jest.Mock).mockImplementation((): object => ({
      ...CuriousityBaseLayoutData
    }));
  });

  it("renders as expected", () => {
    const tree = renderer
      .create(
        <Base>
          <div>
            <span>Test Content</span>
          </div>
        </Base>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
