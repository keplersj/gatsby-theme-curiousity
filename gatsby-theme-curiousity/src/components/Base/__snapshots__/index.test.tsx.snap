// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base Layout renders as expected 1`] = `
<head>
  Object {}
  Object {
    "lang": "en",
  }
  <meta
    content="Test Site"
    data-react-helmet={true}
    property="og:title"
  />
  <meta
    content="https://example.dev/"
    data-react-helmet={true}
    property="og:url"
  />
  <title
    data-react-helmet={true}
  >
    Test Site
  </title>
</head>

<main>
  <div>
    <span>
      Test Content
    </span>
  </div>
</main>
`;
